internal/pkg/service/buffer/statistics/cache/cache.go

```diff
@@ -195,9 +202,12 @@ func prefixForActiveStats(v key.SliceKey) string {
 
 func keyForClosedSlice(v model.Slice) string {
 	switch v.State {
-	case slicestate.Uploading, slicestate.Failed:
+	case slicestate.Uploading:
 		// uploading/<sliceKey>
 		return prefixUploading + v.SliceKey.String()
+	case slicestate.Failed:
+		// failed/<sliceKey>
+		return prefixFailed + v.SliceKey.String()
```

Comment:
- `uploading` slice state znamena, ze pre slice existuje uploading task, a data nahravame do suboru
- `failed` slice state znamena, ze upload spadol a bude retried
- na ucely statistik berieme oba tieto states ako `uploading`, lebo ciel je dostat ich do suboru
- preto som to povodne ukladal aj do rovnakeho cache key
- len tu nastane problem
- Watch API posle pri zmene stavu event (vid `watchClosedSlices` metodu):
```
PUT /slices/active/closed/failed/.....
DEL /slices/active/closed/uploading/.....
```
- a tym padom, ze to cachujem ako rovnaky key, tak najprv si tam nieco ulozim (PUT) no hned si to zmazem (DEL)
- tento case chybal v teste
- doplnil som to tu aj do testu